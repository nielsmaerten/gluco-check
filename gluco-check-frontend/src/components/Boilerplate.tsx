import React, { ReactEventHandler } from "react";
import { Trans, useTranslation } from "react-i18next";
import { Link, makeStyles, Typography } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  legal: {
    "& p": {
      fontSize: ".7rem",
    },
    "& a": {
      fontSize: ".7rem",
      cursor: "pointer",
      color: theme.palette.text.primary,
      textDecoration: "underline",
    },
  },
}));

type BoilerplateProps = {
  handleSignoutClicked?: ReactEventHandler;
};

function Boilerplate({ handleSignoutClicked }: BoilerplateProps) {
  const { t } = useTranslation();
  const classes = useStyles();

  return (
    <>
      <Typography>
        {handleSignoutClicked && (
          <Link variant="body2" color="error" onClick={handleSignoutClicked}>
            {t("boilerplate.logout")}
          </Link>
        )}
      </Typography>
      <div className={classes.legal}>
        <Typography>
          <Trans i18nKey="boilerplate.terms">
            By continuing, you are indicating that you accept our{" "}
            <Link href={t("urls.termsAndConditions")}>Terms of Service</Link>{" "}
            and <Link href={t("urls.privacy")}>Privacy Policy</Link>.
          </Trans>
        </Typography>
        <Typography>{t("boilerplate.google")}</Typography>
        <Typography>
          <Trans i18nKey="boilerplate.nightscout">
            Not affiliated with the{" "}
            <Link href={t("urls.nightscoutProject")}>Nightscout Project</Link>
          </Trans>
        </Typography>
      </div>
    </>
  );
}

export default Boilerplate;
