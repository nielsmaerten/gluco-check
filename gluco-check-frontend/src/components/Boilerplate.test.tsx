import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import Boilerplate from "./Boilerplate";
import userEvent from "@testing-library/user-event";

jest.mock("../lib/firebase.ts", () => {
  return {
    auth: {
      signOut: jest.fn(),
    },
  };
});

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
  Trans: function (props: any) {
    return <span>{props.i18nKey}</span>;
  },
}));

afterEach(cleanup);

describe("Landing page", () => {
  const mockHandleSignoutClicked = jest.fn();
  beforeEach(() => {
    mockHandleSignoutClicked.mockReset();
  });
  it("renders the component", async () => {
    const { container } = render(
      <Boilerplate handleSignoutClicked={mockHandleSignoutClicked} />
    );
    expect(container.firstChild).toMatchSnapshot();

    const logoutButton = await screen.findByText("boilerplate.logout");
    expect(logoutButton).toBeTruthy();

    await userEvent.click(logoutButton);
    expect(mockHandleSignoutClicked).toHaveBeenCalled();
  });

  it("renders the component when not authenticated", () => {
    const { container } = render(<Boilerplate />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(
      <Boilerplate handleSignoutClicked={mockHandleSignoutClicked} />
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});
