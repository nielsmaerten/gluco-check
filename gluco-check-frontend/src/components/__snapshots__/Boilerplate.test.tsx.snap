// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders the component 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1"
>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorError"
  >
    boilerplate.logout
  </a>
</p>
`;

exports[`Landing page renders the component when not authenticated 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1"
/>
`;
